exporter/opentelemetry-exporter-otlp/src/opentelemetry/exporter/otlp/metrics_exporter/__init__.py

```diff
@@ -317,6 +319,19 @@ def _translate_data(
                 )
             )
 
+            instrumentation_library_metrics.instrumentation_library.name = (
+                export_record.instrument.meter.instrumentation_info.name
```

Comment:
should the instrumentation info not be pulled from the resource on the `export_record`?

Reply:
Those values can possible be different I think? I'm wondering what "instrumentation library" in the context of metrics means. For [tracing[(https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/overview.md#instrumentation-libraries) it is instrumentation library (like flask), but here there is no such thing. Whereas [Resource](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/resource/sdk.md#resource-sdk) is defined as the entity that is producing the telemetry. 

We should find out which is the most appropriate for what `otlp` needs in this field.

Reply:
I can't find the instrumentation info in the resource object:

```
ipdb> export_record.resource
<opentelemetry.sdk.resources.Resource object at 0x7f4c390dc730>
ipdb> dir(export_record.resource)
['__class__', '__delattr__', '__dict__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__gt__', '__hash__', '__init__', '__init_subclass__', '__le__', '__lt__', '__module__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', '__weakref__', '_attributes', 'attributes', 'create', 'create_empty', 'merge']
ipdb> export_record.resource
<opentelemetry.sdk.resources.Resource object at 0x7f4c390dc730>
ipdb> export_record.resource.attributes
OrderedDict([('a', 1), ('b', False)])
ipdb> export_record.resource.attributes
                              attributes     create()       create_empty() merge()       
                              instance 
```

This happens because we can have a `Resource` [without those attributes](https://github.com/open-telemetry/opentelemetry-python/pull/1418/files#diff-ce457a78f89c2cae462c4c929d324b8e0afaddc7de525814ec8e8c74ab3a1b82R57), but the `Meter` [will have them](https://github.com/open-telemetry/opentelemetry-python/pull/1418/files#diff-ce457a78f89c2cae462c4c929d324b8e0afaddc7de525814ec8e8c74ab3a1b82R65).

Reply:
> I'm wondering what "instrumentation library" in the context of metrics means. For [tracing[(https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/overview.md#instrumentation-libraries) it is instrumentation library (like flask), but here there is no such thing.

I think it is supposed to be the same thing [from reading this](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/glossary.md#tracer-name--meter-name). I think this code is correct

Reply:
It doesn't look like Observers have a meter, filed a bug https://github.com/open-telemetry/opentelemetry-python/issues/1424